sigstore/oidc.py

```diff
@@ -77,22 +92,28 @@ def __init__(self, raw_token: str) -> None:
         # certificate binding and issuance.
         try:
             self._unverified_claims = jwt.decode(
-                self._raw_token, options={"verify_signature": False}
+                raw_token,
+                options={
+                    "verify_signature": False,
+                    "verify_aud": True,
+                    "verify_iat": True,
+                    "verify_exp": True,
+                    "require": ["aud", "iat", "exp", "iss"],
+                },
+                audience=DEFAULT_AUDIENCE,
             )
-        except jwt.InvalidTokenError as exc:
-            raise IdentityError("invalid identity token") from exc
+        except Exception as exc:
+            raise IdentityError(
+                "Identity token is malformed or missing claims"
+            ) from exc
```

Comment:
The actual traceback _should_ still contain that specificity, because of the `from exc` -- the traceback chain will contain the underlying `pwjwt` exception, which in turn (should) mention the missing or invalid claim.

(I can make that more explicit by embedding `{exc}` rather than chaining with it, if you'd like.)

Reply:
I don't love depending on the exception from `pyjwt`, but let's see what happens.